### PR TITLE
test(integration)_: separate anvil from tests

### LINKED ISSUE #15118: CI for status-go integration tests
* rework contract deployments to checkout and build a container instead of mounting volumes with contract repos
* add make target for status go anvil + foundry + status-go as daemon and tests
* add make target for status go anvil + foundry
* add Jenkinsfile to run tests and report results under every commit in every PR of status-go

### PR DESCRIPTION
**Pull request by antdanchenko**
separate anvil from tests

* anvil are separated from tests and can be used in development instead of infura/pokt RPC
* added make targets
* added Dockerfile to clone and deploy contracts, as result there is no need to do that manually 

Important changes:
- [x] not sure if that is right way to add make targets but it works

[15118](https://github.com/status-im/status-desktop/issues/15118)

### CODE REVIEW

**Review by yevh-berdnyk — APPROVED** [review]

File: integration-tests/README.MD [hunk 1]
```diff
@@ -13,22 +13,23 @@ Integration tests for status-go
...
-* Add environment variable, path to contract repos e.g. `export PATH_TO_CONTRACT_REPOS=~/directory-with-every-contract-repo`
 * Install [Python 3.10.14](https://www.python.org/downloads/)
 * In `integration-tests` directory, run `pip install -r requirements.txt`
 * **Optional (for test development)**: Use Python virtual environment for better dependency management. You can follow the guide [here](https://akrabat.com/creating-virtual-environments-with-pyenv/):
 
 ## How to Run
 
-- In `integration-tests/tests` run `docker compose up --build`, as result:
+### Running dev RPC (anvil with contracts)
+- In `integration-tests/tests` run `docker compose -f docker-compose.anvil.yml up --remove-orphans --build`, as result:
```
Comment (yevh-berdnyk) [inline integration-tests/README.MD]: I guess it should be 
```suggestion
- In `integration-tests` run `docker compose -f docker-compose.anvil.yml up --remove-orphans --build`, as result:
```
because docker-compose.anvil.yml is not in the `tests` directory

**Review by anastasiyaig — APPROVED** [review]
the issue with status-go crash will be addressed separately (or maybe in this PR)

File: Makefile [hunk 2]
```diff
@@ -477,3 +477,9 @@ build-verif-proxy-wrapper:
 
 test-verif-proxy-wrapper:
 	CGO_CFLAGS="$(CGO_CFLAGS)" go test -v github.com/status-im/status-go/rpc -tags gowaku_skip_migrations,nimbus_light_client -run ^TestProxySuite$$ -testify.m TestRun -ldflags $(LDFLAGS)
+
+run-integration-tests:
```
Comment (jakubgs) [inline Makefile:482]:
```suggestion
run-integration-tests: SHELL := /bin/sh
run-integration-tests:
```

File: Makefile [hunk 3]
```diff
@@ -477,3 +477,9 @@ build-verif-proxy-wrapper:
...
+	docker-compose -f integration-tests/docker-compose.anvil.yml -f integration-tests/docker-compose.test.status-go.yml up --remove-orphans --build
+
+run-anvil:
```
Comment (jakubgs) [inline Makefile:486]:
```suggestion
run-anvil: SHELL := /bin/sh
run-anvil:
```

File: integration-tests/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    apk add git
+
+WORKDIR /app
+
+RUN git clone https://github.com/$github_org/$github_repo
+
+WORKDIR /app/$github_repo
+
+RUN git submodule deinit --force .
+RUN git submodule update --init --recursive
```
Comment (jakubgs) [inline integration-tests/Dockerfile]: Won't this just create a lot of very big Docker images? Why can't we do this at image runtime?

File: integration-tests/docker-compose.anvil.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        github_org: status-im
+        github_repo: status-network-token-v2
+    entrypoint: ["forge", "script", "script/Deploy.s.sol", "--broadcast", "--fork-url", "http://anvil:8545", "--private-key=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"]
+
+  deploy-communities-contracts:
+    platform: linux/amd64
+    environment:
+      - API_KEY_ETHERSCAN=""
+      - API_KEY_ARBISCAN=""
+      - API_KEY_OPTIMISTIC_ETHERSCAN=""
```
Comment (jakubgs) [inline integration-tests/docker-compose.anvil.yml]: Why are we setting this to nothing? Wouldn't it make more sense just to skip this whole section? Or does setting them to empty strings does something specific? And if it does, it probably should be commented upon.

**Review by jakubgs — CHANGES_REQUESTED** [review]

File: integration-tests/docker-compose.anvil.yml (hunk 5 above)
Comment (antdanchenko) [reply integration-tests/docker-compose.anvil.yml]: it does, having env var is required (crashes without) but values aren't used. will add comments

File: integration-tests/docker-compose.anvil.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      - API_KEY_ETHERSCAN="" # env var is required, but value isn't used
+    depends_on:
+      - anvil
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        github_org: status-im
+        github_repo: status-network-token-v2
+    command: ["forge", "script", "script/Deploy.s.sol", "--broadcast", "--fork-url", "http://anvil:8545", "--private-key=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"]
```
Comment (jakubgs) [inline integration-tests/docker-compose.anvil.yml]: Long lines like this are hard to follow:
```suggestion
    command: |
        forge script script/Deploy.s.sol
            --broadcast
            --fork-url=http://anvil:8545
            --private-key=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```
Same for the other line.

File: integration-tests/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,18 @@
+FROM ghcr.io/foundry-rs/foundry:latest as checkout
+
+ARG github_repo
+ARG github_org
+
+ENV GITHUB_REPO=$github_repo
+ENV GITHUB_ORG=$github_org
```
Comment (jakubgs) [inline integration-tests/Dockerfile]: You can just use `ENV`, or even skip it entirely here. `ARG` are for build-time, but you want ot use those at runtime. Please see:
https://vsupalov.com/docker-arg-env-variable-guide/#arg-and-env

And just use `environment` section in the Docker Compose.

**Review by jakubgs — APPROVED** [review]

**Review by jakubgs — APPROVED** [review]
